scalding-core/src/main/scala/com/twitter/scalding/TypedPipeChecker.scala

```diff
@@ -0,0 +1,24 @@
+package com.twitter.scalding
+
+/**
+ * This class is used to assist with testing a TypedPipe
+ */
+object TypedPipeChecker {
+  /*
+   * Execute a TypedPipe in memory, convert the resulting Iterator to
+   * a list and run it through a function that makes arbitrary
+   * assertions on it.
+   */
+  def checkOutput[T](output: TypedPipe[T])(assertions: List[T] => Unit) =
```

Comment:
Is it worth having a flavor like
```
def checkOutputTransform[T, U](input: List[T])(transform: TypedPipe[T] => TypedPipe[U])(assertions: List[T] => Unit) =
  assertions(checkOutputInline(transform(TypedPipe.from(input))))
```

Or doing TypedPipe.from we should leave to the test writer?